File: frontend/src/main/web/src/app/model/search-result.ts
import {Movie} from "./movie";

export class SearchResult {
  private _Search: Array<Movie>;
  private _totalResults: number;
  private _Response: string;
  private _Error: string;
  private _fromDB: boolean;

  get Search(): Array<Movie> {
    return this._Search;
  }

  set Search(value: Array<Movie>) {
    this._Search = value;
  }

  get totalResults(): number {
    return this._totalResults;
  }

  set totalResults(value: number) {
    this._totalResults = value;
  }

  get Response(): string {
    return this._Response;
  }

  set Response(value: string) {
    this._Response = value;
  }

  get Error(): string {
    return this._Error;
  }

  set Error(value: string) {
    this._Error = value;
  }

  get fromDB(): boolean {
    return this._fromDB;
  }

  set fromDB(value: boolean) {
    this._fromDB = value;
  }
}

File: backend/src/main/java/app/model/SearchResult.java
package app.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class SearchResult {

    @JsonProperty(value = "Search")
    private List<Movie> movies;

    @JsonProperty("totalResults")
    private Integer totalResults;

    @JsonProperty("Response")
    private String response;

    @JsonProperty("Error")
    private String error;

    @JsonInclude
    private boolean fromDB;

    public List<Movie> getMovies() {
        return movies;
    }

    public void setMovies(List<Movie> movies) {
        this.movies = movies;
    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }

    public String getResponse() {
        return response;
    }

    public void setResponse(String response) {
        this.response = response;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public boolean isFromDB() {
        return fromDB;
    }

    public void setFromDB(boolean fromDB) {
        this.fromDB = fromDB;
    }
}
